src/prpy/serialization.py

```diff
@@ -99,32 +101,51 @@ def serialize_environment_file(env, path, writer=None):
 
     return data
 
-def serialize_kinbody(body):
-    all_joints = []
-    all_joints.extend(body.GetJoints())
-    all_joints.extend(body.GetPassiveJoints())
+def serialize_kinbody(body, uri_only=False):
 
+    uri = body.GetXMLFilename()
+    
+    if uri_only and not uri:
+        serialization_logger.warn(
+            'uri_only passed, but kinbody {} doesn\'t have a URI.'.format(
+            body.GetName()))
+        uri_only = False
+    
     data = {
         'is_robot': body.IsRobot(),
         'name': body.GetName(),
-        'uri': body.GetXMLFilename(),
-        'links': map(serialize_link, body.GetLinks()),
-        'joints': map(serialize_joint, all_joints),
+        'uri': uri,
     }
-    data['kinbody_state'] = serialize_kinbody_state(body)
+
+    # only add uri_only to serialization so that older deserializers work
+    if uri_only:
+        data['uri_only'] = True
+    
+    if not uri_only:
+        all_joints = []
+        all_joints.extend(body.GetJoints())
+        all_joints.extend(body.GetPassiveJoints())
+        data['links'] = map(serialize_link, body.GetLinks())
+        data['joints'] = map(serialize_joint, all_joints)
+    
+    data['kinbody_state'] = serialize_kinbody_state(body, uri_only=uri_only)
 
     if body.IsRobot():
-        data.update(serialize_robot(body))
+        data.update(serialize_robot(body, uri_only=uri_only))
 
     return data
 
-def serialize_robot(robot):
-    return {
-        'manipulators': map(serialize_manipulator, robot.GetManipulators()),
+def serialize_robot(robot, uri_only=False):
+    data = {
         'robot_state': serialize_robot_state(robot),
     }
+    if not uri_only:
+        data.update({
+            'manipulators': map(serialize_manipulator, robot.GetManipulators()),
+        })
+    return data
 
-def serialize_kinbody_state(body):
+def serialize_kinbody_state(body, uri_only=False):
```

Comment:
This function no longer needs the `uri_only` parameter. It doesn't make sense conceptually, either, since the state must be serialized even if `uri_only` is `True`.